### PR TITLE
Updating PR CI to run full native if no changes in modules are detected

### PR DESCRIPTION
**Pull request by jedla97**
### Summary

Adding the logic to run the Linux native run if no modules are detected. This change not include windows as on PRs we don't even run native windows.

I decided make `NATIVE_MODULES_MAVEN_PARAM`  on multiple row as it's more readable. So I need to clean it with `tr -d -s '\n' ' '` for env variable to be one line string.

Also updating the modules load for native execution time be similar for all modules.

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [x] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by jedla97** [comment]
Flaky test is `io.quarkus.ts.messaging.kafka.reactive.streams.KafkaGratefulShutdownIT.testConnection`

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -167,12 +173,13 @@ jobs:
   linux-build-native-latest:
     name: PR - Linux - Native build - Latest Version
     runs-on: ubuntu-latest
-    needs: detect-test-suite-modules
+    needs: prepare-jvm-native-latest-modules-mvn-param
     env:
       MODULES_ARG: ${{ needs.detect-test-suite-modules.outputs.MODULES_ARG }}
```
Comment (gtroitsk) [inline .github/workflows/ci.yml:178]: I think `MODULES_ARG` can be removed now

Comment (jedla97) [reply .github/workflows/ci.yml]: Thanks missed this! Updated.

**Comment by rsvoboda** [comment]
Ready to merge?

**Comment by gtroitsk** [comment]
> Ready to merge?

yes, merged

**Pull request merged**